rust/agama-dbus-server/src/network/model.rs

```diff
@@ -306,24 +306,96 @@ impl Connection {
     pub fn is_loopback(&self) -> bool {
         matches!(self, Connection::Loopback(_))
     }
+
+    pub fn is_ethernet(&self) -> bool {
+        matches!(self, Connection::Loopback(_)) || matches!(self, Connection::Ethernet(_))
+    }
 }
 
 #[derive(Debug, Default, Clone)]
 pub struct BaseConnection {
     pub id: String,
     pub uuid: Uuid,
+    pub mac_address: Option<MacAddress>,
     pub ip_config: IpConfig,
     pub status: Status,
     pub interface: String,
     pub match_config: MatchConfig,
 }
 
+impl BaseConnection {
+    pub fn mac_address_string(&self) -> String {
+        match &self.mac_address {
+            Some(mac) => mac.to_string(),
+            None => "".to_string(),
+        }
+    }
+
+    pub fn set_mac_address(&mut self, mac_address: &str) -> Result<(), InvalidMacAddress> {
```

Comment:
This one looks counter-intuitive to me. Why do I need to convert a MacAddress to a string? Not to mention the fact that an empty string means `None`. If you add the `None` variant to the `MacAddress` enum, well, problem solved.

If you continue with the `Option` approach, I would 1) pass an option or 2) have a `set_mac_address` and a `unset_mac_address`.